src/OutputWriters/jld2_output_writer.jl

```diff
@@ -26,41 +28,102 @@ end
-  and write to a new one with a consistent naming scheme ending in `part1`, `part2`, etc. Defaults to `Inf`.
-- `force`: Remove existing files if their filenames conflict. Default: `false`.
-- `async`: Write output asynchronously. Default: `false`.
+                  and write to a new one with a consistent naming scheme ending in `part1`, `part2`, etc.
+                  Defaults to `Inf`.
+
+- `force`: Remove existing files if their filenames conflict.
+           Default: `false`.
+
+- `init`: A function of the form `init(file, model)` that runs when a JLD2 output file is initialized.
+          Default: `noinit(args...) = nothing`.
+
+- `async`: Write output asynchronously.
+           Default: `false`.
+
 - `verbose`: Log what the output writer is doing with statistics on compute/write times and file sizes.
-  Default: `false`.
+             Default: `false`.
+
+- `including`: List of model properties to save with every file.
+               Default: `[:grid, :coriolis, :buoyancy, :closure]`
+
+- `part`: The starting part number used if `max_filesize` is finite.
+          Default: 1.
+
 - `jld2_kw`: Dict of kwargs to be passed to `jldopen` when data is written.
 """
-function JLD2OutputWriter(model, outputs; iteration_interval=nothing, time_interval=nothing,
-                          dir=".", prefix="", init=noinit,
-                          including=[:grid, :coriolis, :buoyancy, :closure],
-                          part=1, max_filesize=Inf, force=false, async=false, verbose=false,
-                          jld2_kw=Dict{Symbol, Any}())
+function JLD2OutputWriter(model, outputs; prefix,
+                                                            dir = ".",
+                                             iteration_interval = nothing,
+                                                  time_interval = nothing,
+                                          time_averaging_window = nothing,
+                                          time_averaging_stride = 1,
+                                                   max_filesize = Inf,
+                                                          force = false,
+                                                           init = noinit,
+                                                          async = false,
+                                                        verbose = false,
+                                                      including = [:grid, :coriolis, :buoyancy, :closure],
+                                                           part = 1,
+                                                        jld2_kw = Dict{Symbol, Any}())
 
     validate_intervals(iteration_interval, time_interval)
 
+    # Convert each output to WindowedTimeAverage if time_averaging_window is specified
+    if !isnothing(time_averaging_window)
+
+        !isnothing(iteration_interval) && error("Cannot specify iteration_interval with time_averaging_window.")
```

Comment:
I might throw an `ArgumentError` as it's more specific and allows us to `@test_throws` the constructor.